clarifai/runners/models/model_run_locally.py

```diff
@@ -0,0 +1,192 @@
+    ]
+
+    #  Ensure there is exactly one subclass of BaseRunner in the model.py file
+    if len(classes) != 1:
+      raise Exception("Expected exactly one subclass of BaseRunner, found: {}".format(
+          len(classes)))
+
+    MyRunner = classes[0]
+    return MyRunner
+
+  def _build_request(self):
+    """Create a mock inference request for testing the model."""
+
+    return service_pb2.PostModelOutputsRequest(
+        model=resources_pb2.Model(model_version=resources_pb2.ModelVersion(
+            id="model_version", output_info=resources_pb2.OutputInfo())),
+        inputs=[
+            resources_pb2.Input(data=resources_pb2.Data(
+                text=resources_pb2.Text(raw="How many people live in new york?"),
+                image=resources_pb2.Image(url="https://samples.clarifai.com/metro-north.jpg"),
+                audio=resources_pb2.Audio(url="https://samples.clarifai.com/GoodMorning.wav"),
+            ))
+        ],
+    )
+
+  def _run_model_inference(self, runner):
+    """Perform inference using the runner."""
+    request = self._build_request()
+
+    try:
+      return runner.predict(request)
+    except Exception as e:
+      logger.error(f"Model Prediction failed: {e}")
+      traceback.print_exc()
+      return service_pb2.MultiOutputResponse(status=status_pb2.Status(
+          code=status_code_pb2.MODEL_PREDICTION_FAILED,
+          description="Prediction failed",
+          details="",
+          internal_details=str(e),
+      ))
+
+  def _run_test(self):
+    """Test the model locally by making a prediction."""
+    # validate that we have checkpoints downloaded before constructing MyRunner
+    uploader = ModelUploader(self.model_path)
+    uploader.download_checkpoints()
+
+    # construct MyRunner which will call load_model()
+    MyRunner = self._get_model_runner()
+    runner = MyRunner(
+        runner_id="n/a",
+        nodepool_id="n/a",
+        compute_cluster_id="n/a",
+    )
+    runner.load_model()
+
+    # send an inference.
+    response = self._run_model_inference(runner)
+    if response.outputs[0].status.code != status_code_pb2.SUCCESS:
+      logger.error(f"Moddel Prediction failed: {response}")
+    else:
+      logger.info(f"Model Prediction succeeded: {response}")
+
+  def test_model(self):
+    """Test the model by running it locally in the virtual environment."""
+    command = [
+        self.python_executable,
+        "-c",
+        f"import sys; sys.path.append('{os.path.dirname(os.path.abspath(__file__))}'); "
+        f"from model_run_locally import ModelRunLocally; ModelRunLocally('{self.model_path}')._run_test()",
+    ]
+    try:
+      logger.info("Testing the model locally...")
+      subprocess.check_call(command)
+      logger.info("Model tested successfully!")
+    except subprocess.CalledProcessError as e:
+      logger.error(f"Error testing the model: {e}")
+      sys.exit(1)
+
+  # run the model server
+  def run_model_server(self):
```

Comment:
I include this method to start the model runner server locally, but not sure if's useful here